File: src/main/java/com/machfour/macros/objects/RegularMeal.java
package com.machfour.macros.objects;

import com.machfour.macros.core.*;

public class RegularMeal extends MacrosEntity<RegularMeal> {

    public RegularMeal(ColumnData<RegularMeal> data, ObjectSource objectSource) {
        super(data, objectSource);
    }

    @Override
    public Table<RegularMeal> getTable() {
        return Schema.RegularMealTable.instance();
    }

    public static Factory<RegularMeal> factory() {
        return RegularMeal::new;
    }
    @Override
    public Factory<RegularMeal> getFactory() {
        return factory();
    }
}

File: src/main/java/com/machfour/macros/objects/RegularMeal.kt
package com.machfour.macros.objects

import com.machfour.macros.core.*
import com.machfour.macros.core.schema.RegularMealTable
import com.machfour.macros.core.schema.SchemaHelpers

class RegularMeal(data: ColumnData<RegularMeal>, objectSource: ObjectSource) : MacrosEntityImpl<RegularMeal>(data, objectSource) {
    companion object {
        val table: Table<RegularMeal>
            get() = RegularMealTable.instance
        val factory: Factory<RegularMeal> = Factory { dataMap, objectSource -> RegularMeal(dataMap, objectSource) }
    }

    override val factory: Factory<RegularMeal>
        get() = Companion.factory

    override val table: Table<RegularMeal>
        get() = Companion.table

}
